backend/src/main/java/com/bakdata/conquery/models/query/queryplan/SecondaryIdQueryPlan.java

```diff
@@ -152,18 +115,32 @@ private void executeQueriesWithSecondaryId(QueryExecutionContext ctx, Entity ent
 				continue;
 			}
 
-			int start = bucket.getEntityStart(entityId);
-			int end = bucket.getEntityEnd(entityId);
+			final int start = bucket.getEntityStart(entityId);
+			final int end = bucket.getEntityEnd(entityId);
 
 			for (int event = start; event < end; event++) {
 				//we ignore events with no value in the secondaryIdColumn
 				if (!bucket.has(event, secondaryIdColumnId)) {
 					continue;
 				}
 
-				String key = ((String) bucket.createScriptValue(event, secondaryIdColumnId));
-				final ConceptQueryPlan plan = childPerKey.computeIfAbsent(key, k -> createChild(ctxWithPhase, bucket));
-				plan.nextEvent(bucket, event);
+				final String key = ((String) bucket.createScriptValue(event, secondaryIdColumnId));
+
+				if (childPerKey.containsKey(key)) {
+					final ConceptQueryPlan plan = childPerKey.get(key);
+					plan.nextEvent(bucket, event);
+				}
+				else {
+					final ConceptQueryPlan plan = createChild(ctxWithPhase, bucket);
+					final boolean consumed = plan.nextEvent(bucket, event);
+
+					if (consumed) {
+						childPerKey.put(key, plan);
+					}
+					else {
+						childPlanReusePool.add(plan);
+					}
+				}
```

Comment:
enthält `childPlanReusePool` nicht für jedes nicht konsumierte Event duplikate? Und wenn dann doch ein Event Konsumiert wird, dann gibt es auch noch ein Duplikat in `childPerKey`

Reply:
Pläne aus dem childPlanReusePool werden vor dem verwenden in createChild gecleart. 

---

Ich verstehe nicht ganz was der kommentar bedeuten soll ehrlich gesagt. Entweder wurde das event konsumiert, dann behalten wir den Plan, oder nicht, dann verwerfen wir ihn wieder.

Reply:
Okay, jetzt habe ich erst die Rotation der pläne durch schaut